src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -0,0 +1,409 @@
+ * values of epsilon.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffEpsilonTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 1.0);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that the cluster assignment is different for different
+ * values of Min Size.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffMinSizeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.4);
+  SetInputParam("min_size", (int) 5);
+
+  mlpackMain();
+
+  arma::Row<size_t> output1;
+  output1 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  bindings::tests::CleanMemory();
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["epsilon"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["min_size"].wasPassed = false;
+
+  SetInputParam("input", inputData);
+  SetInputParam("epsilon", (double) 0.5);
+  SetInputParam("min_size", (int) 40);
+
+  mlpackMain();
+
+  arma::Row<size_t> output2;
+  output2 = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
+  BOOST_REQUIRE_GT(arma::accu(output1 != output2), 1);
+}
+
+/**
+ * Check that the tree type should be from the given list of
+ * tree types. ’kd’, ’r’, ’r-star’, ’x’, ’hilbert-r’, ’r-plus’,
+ * ’r-plus-plus’, ’cover’, ’ball’.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANTreeTypeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("tree_type", std::string("binary"));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check that the assignment of cluster is same if
+ * different tree type is used for search.
+ */
+BOOST_AUTO_TEST_CASE(DBSCANDiffTreeTypeTest)
+{
+  arma::mat inputData;
+  if (!data::Load("iris.csv", inputData))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+
+  SetInputParam("input", inputData);
+  SetInputParam("tree_type", std::string("kd"));
+
+  mlpackMain();
+
+  arma::Row<size_t> kdOutput;
+  kdOutput = std::move(CLI::GetParam<arma::Row<size_t>>("assignments"));
+
```

Comment:
Maybe you can add some comments to differentiate the start/end for different trees. Sorry, for being so picky. It's your call if you like to, don't feel obligated.